src/main/java/seedu/module/model/module/predicate/SameModuleCodePredicate.java

```diff
@@ -17,6 +17,7 @@ public SameModuleCodePredicate(String moduleCode) {
 
     @Override
     public boolean test(Module module) {
+        assert !module.equals("") : "SameModuleCodePredicate received an empty String.";
```

Comment:
This is a redundant assertion.

Reply:
The assertion is there in case in the future, there might be methods that use this class which can potentially pass in an empty string. Passing in an empty string here will cause an error to be thrown.

Reply:
But it's a Module..

Reply:
Oh right, my bad asserted the wrong object.